react-carousel-component/src/carousel.jsx

```diff
@@ -0,0 +1,110 @@
+import React from 'react';
+
+const images = [
+  {
+    src: '001.png',
+    alt: 'Bulbasaur'
+  },
+  {
+    src: '004.png',
+    alt: 'Charmander'
+  },
+  {
+    src: '007.png',
+    alt: 'Squirtle'
+  },
+  {
+    src: '025.png',
+    alt: 'Pikachu'
+  },
+  {
+    src: '039.png',
+    alt: 'Jigglypuff'
+  }
+];
+
+let intervalId;
+
+class Carousel extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      imageIndex: 0
+    };
+    this.handleClick = this.handleClick.bind(this);
+    this.carouselChange = this.carouselChange.bind(this);
+  }
+
+  handleClick(event) {
+    clearInterval(intervalId);
+    let index;
+    if (event.target.getAttribute('data-view') === 'right') {
+      index = this.state.imageIndex === images.length - 1
+        ? 0
+        : this.state.imageIndex + 1;
+    } else if (event.target.getAttribute('data-view') === 'left') {
+      index = this.state.imageIndex === 0
+        ? images.length - 1
+        : this.state.imageIndex - 1;
+    } else {
+      index = Number(event.target.getAttribute('data-view'));
+    }
+    this.setState({ imageIndex: index });
+    intervalId = setInterval(this.carouselChange, 3000);
```

Comment:
Store your `intervalId` on `this` so that your component can be instantiated more than once.